### PR TITLE
feat(formula): add config

### LINKED ISSUE #777: Allow publishing under bin name instead of package name
Hello!

We want to publish our binary on homebrew under the binary name, but it is published under the crate name. Is there a way to do that?

### PR DESCRIPTION
**Pull request by ashleygwilliams**
fixes #777

### CODE REVIEW

File: cargo-dist/tests/integration-tests.rs [hunk 1]
```diff
@@ -62,6 +62,42 @@ path-guid = "BFD25009-65A4-4D1E-97F1-0030465D90D6"
...
+fn axolotlsay_custom_formula() -> Result<(), miette::Report> {
+    let test_name = _function_name!();
+    AXOLOTLSAY.run_test(|ctx| {
+        let dist_version = ctx.tools.cargo_dist.version().unwrap();
+        ctx.patch_cargo_toml(format!(r#"
+[workspace.metadata.dist]
+cargo-dist-version = "{dist_version}"
+installers = ["homebrew"]
+tap = "axodotdev/homebrew-packages"
+formula = "AXOLOTLSHOUT"
```
Comment (mistydemeo) [inline cargo-dist/tests/integration-tests.rs:75]: We probably want to test with a lowercase name, not an uppercase name. Capitalization is significant and has some implications.

We could also consider normalizing the name based on other factors in a later update, but I think it also just makes sense to link people to the Homebrew docs about formula names when they're picking one.

**Review by mistydemeo — DISMISSED** [review]

**Comment by mistydemeo** [comment]
I'm seeing a couple things here we want to make sure we fix:

* The snapshots aren't capturing the formula if the name is specified - not sure what's going on yet.
* Running locally, I'm seeing that the wrong class name is generated in the formula; the culprit is this line, which is still keyed off app name: https://github.com/axodotdev/cargo-dist/blob/d19b47f37467cdc4b5040c7dfa59c9e7964782b9/cargo-dist/src/tasks.rs#L1888

**Comment by mistydemeo** [comment]
Realized that the snapshots are missing the formula because the name to add to the snapshot is hardcoded: https://github.com/axodotdev/cargo-dist/blob/main/cargo-dist/tests/gallery/dist.rs#L214

It's probably easiest to turn this into a glob that grabs all `*.rb` files.

**Review by MarinPostma — COMMENTED** [review]
Looks like exactly what I need, thanks @ashleygwilliams

**Comment by ashleygwilliams** [comment]
add docs

File: cargo-dist/tests/gallery/dist.rs [hunk 2]
```diff
@@ -209,21 +209,51 @@ impl<'a> TestContext<'a, Tools> {
...
+        assert!(
+            number_found <= 1,
+            "found {} files with the extension {}, expected 1 or 0",
+            number_found,
+            extension
+        );
+        files.first().cloned()
+    }
+
+    fn load_files_with_extension(dirname: Utf8PathBuf, extension: &str) -> Vec<Utf8PathBuf> {
```
Comment (Gankra) [inline cargo-dist/tests/gallery/dist.rs:240]: I suggest

```suggestion
    fn find_files_with_suffix(dirname: &Utf8Path, suffix: &str) -> Vec<Utf8PathBuf> {
```

(and a bunch of fallout changes from that)

**Comment by Gankra** [comment]
condolences, merged changes need this pr rebased and the snapshot regen'd

**Review by Gankra — COMMENTED** [review]
test harness changes look good

i was sure runtest_homebrew_installer was gonna need edits but nope, it was already perfectly robust to only assuming it had the path it was given!

File: book/src/reference/config.md [hunk 3]
```diff
@@ -345,6 +345,17 @@ For instance for packages that are a library and a CLI binary, some developers p
...
+
+> since 0.11.0
+
+Example: `formula = "axolotlbrew"`
+
+Specifies a string to override the default Homebrew formula name (the app name). This works
+well specifically for folks who are customizing their bin name and would like the Homebrew
+formula to match the bin name as opposed to the app name (which, in Rust, is the crate name).
+
+You must set this on `[package.metadata.dist]` and not `[workspace.metadata.dist]`.
```
Comment (Gankra) [inline book/src/reference/config.md:358]: heh, good catch.

**Review by Gankra — APPROVED** [review]

File: book/src/installers/homebrew.md [hunk 4]
```diff
@@ -9,6 +9,27 @@ cargo-dist can, optionally, publish your formula to a tap repository for you on
 ```toml
 tap = "axodotdev/homebrew-formulae"
 publish-jobs = ["homebrew"]
+
+Since 0.11.0, cargo-dist can, optionally, also customize your Homebrew formula name.
+By default, your formula will be named using the app name (in Rust, this is the crate
+name). If you are overriding the bin name, you may want to make your Homebrew formula
+match- you can do so with config like this:
+
+```toml
```
Comment (Gankra) [inline book/src/installers/homebrew.md:18]: uhhh did something go wrong here

Comment (ashleygwilliams) [reply book/src/installers/homebrew.md:19]: yes, apologies, forcepushed fix

**Review by Gankra — APPROVED** [review]

**Pull request merged**